### DISCUSSION TITLE
Oracles sources

### DISCUSSION THREAD

**Discussion by Defianer** [DeFiCh/app#1158] [Q&A]
Hi team,

would it be possible to publish all Oracle IDs and for each ID the source?

I'm aware the sources can be found here: https://github.com/DeFiCh/oracle-adapters/tree/main/adapters/src
However, this doesn't allow finding the 1:1 relationship between Oracle ID and the source.

Would be great to get this transparency!
One reason I'm asking is because it's less predictable/ annoying, when Oracles publish outdated data.
It's 3:15pm CET now, markets are open, and yet e.g. for dBABA, oracle IDs 4367AEE8 and 55D59E44 are still publishing outdated prices for BABA, which are the closing prices of the market yesterday, but not live quotes.

Phil

**Comment by TatyOko28** [comment]
Hi Phil,

Thank you for reaching out with your request regarding Oracle IDs and their corresponding sources.

Currently, while the sources for Oracle data are available in the [oracle-adapters repository](https://github.com/DeFiCh/oracle-adapters/tree/main/adapters/src), we acknowledge that there is no direct 1:1 mapping publicly available between Oracle IDs and their respective data sources.

To improve transparency, we will explore the possibility of publishing a full list of Oracle IDs along with their respective sources. This would help provide more predictability regarding data origins and prevent issues such as outdated prices being published.

Regarding the specific issue you highlighted with dBABA and Oracle IDs 4367AEE8 and 55D59E44, we understand that receiving outdated prices during active market hours can be problematic. We are currently investigating why these Oracles are still publishing the previous day's closing prices instead of live market quotes. If this is a recurring issue, we will consider implementing mechanisms to improve data freshness.

We appreciate your patience and your feedback. We'll keep you posted on any updates regarding Oracle source transparency.
